components/script/dom/bindings/js.rs

```diff
@@ -437,6 +437,13 @@ impl<T: Reflectable> LayoutJS<T> {
         debug_assert!(thread_state::get().is_layout());
         *self.ptr
     }
+
+    /// Returns a reference to the interior of this JS object. This method is
+    /// safe to call because it originates from the layout thread, and it cannot
+    /// mutate DOM nodes.
```

Comment:
On the contrary, this should always be only called by the script thread, hence the `for_script` prefix. But you're right that I need an assertion here.